test/common/shared_pool/shared_pool_test.cc

```diff
@@ -174,5 +174,47 @@ TEST_F(SharedPoolTest, RaceCondtionForGetObjectWithObjectDeleter) {
   deferredDeleteSharedPoolOnMainThread(pool);
 }
 
+TEST_F(SharedPoolTest, HashCollision) {
+  Event::MockDispatcher dispatcher;
+  struct MyHash {
+    constexpr size_t operator()(int x) const { return x < 10 ? 0 : 1; }
+  };
+
+  auto pool = std::make_shared<ObjectSharedPool<int, MyHash>>(dispatcher);
+  {
+    // Verify that the hash function works as intended.
+    static_assert(MyHash{}(4) == 0);
+    static_assert(MyHash{}(3) == 0);
+    static_assert(MyHash{}(15) == 1);
+    static_assert(MyHash{}(12) == 1);
+
+    // Instantiate objects that hash to the same value.
+    auto o = pool->getObject(4);
+    auto o1 = pool->getObject(3);
+
+    // Verify that there are separate entries in the pool for objects with the
+    // same hash value.
+    ASSERT_EQ(2, pool->poolSize());
```

Comment:
these should all be EXPECT_EQ I think. I only use ASSERT_EQ if a failure should end the test, like ASSERT_EQ that an array size is expected prior to accessing array elements, which would crash if the size was not as expected.

Reply:
Thanks, makes sense.

Reply:
so are you going to change it?

Reply:
Yes, sorry.